internal/commands/accesslist/delete_test.go

```diff
@@ -0,0 +1,243 @@
+		{ID: "address2", Address: "192.1.1.1"},
+		{ID: "address3", Address: "192.158.1.38", Comment: "cool comment"},
+	}
+
+	t.Run("should show empty state message if no allowed ips are found", func(t *testing.T) {
+		_, ui := mock.NewUI()
+
+		realmClient := mock.RealmClient{}
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			return []realm.App{app}, nil
+		}
+		realmClient.AllowedIPsFn = func(groupID, appID string) ([]realm.AllowedIP, error) {
+			return nil, nil
+		}
+
+		cmd := &CommandDelete{deleteInputs{
+			ProjectInputs: cli.ProjectInputs{
+				Project: projectID,
+				App:     appID,
+			},
+		}}
+
+		assert.Equal(t, errors.New("No IP addresses or CIDR blocks to delete"), cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+	})
+
+	for _, tc := range []struct {
+		description    string
+		testInput      []string
+		expectedOutput string
+		deleteErr      error
+	}{
+		{
+			description: "should return successful outputs for proper inputs",
+			testInput:   []string{"0.0.0.0", "192.1.1.1"},
+			expectedOutput: strings.Join([]string{
+				"Deleted 2 IP address(es) and/or CIDR block(s)",
+				"  IP Address  Comment  Deleted  Details",
+				"  ----------  -------  -------  -------",
+				"  0.0.0.0     comment  true            ",
+				"  192.1.1.1            true            ",
+				"",
+			}, "\n"),
+		},
+		{
+			description: "should output the errors for deletes on individual allowed ips",
+			testInput:   []string{"0.0.0.0", "192.158.1.38"},
+			expectedOutput: strings.Join([]string{
+				"Deleted 2 IP address(es) and/or CIDR block(s)",
+				"  IP Address    Comment       Deleted  Details           ",
+				"  ------------  ------------  -------  ------------------",
+				"  0.0.0.0       comment       false    something happened",
+				"  192.158.1.38  cool comment  false    something happened",
+				"",
+			}, "\n"),
+			deleteErr: errors.New("something happened"),
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			realmClient := mock.RealmClient{}
+
+			var deleteArgs struct {
+				groupID, appID, allowedIPID string
+			}
+
+			realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				return []realm.App{app}, nil
+			}
+			realmClient.AllowedIPsFn = func(groupID, appID string) ([]realm.AllowedIP, error) {
+				return allowedIPs, nil
+			}
+			realmClient.AllowedIPDeleteFn = func(groupID, appID, allowedIPID string) error {
+				deleteArgs = struct {
+					groupID, appID, allowedIPID string
+				}{groupID, appID, allowedIPID}
+
+				return tc.deleteErr
+			}
+
+			cmd := &CommandDelete{deleteInputs{
+				cli.ProjectInputs{projectID, appID, nil},
+				tc.testInput,
+			}}
+
+			out, ui := mock.NewUI()
+
+			assert.Nil(t, cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+			assert.Equal(t, tc.expectedOutput, out.String())
+			assert.Equal(t, "projectID", deleteArgs.groupID)
+			assert.Equal(t, "appID", deleteArgs.appID)
```

Comment:
can we assert on the `allowedIPID` thats passed in somehow?

Reply:
since we're passing in multiple different `allowedIPID`'s, I wasn't too sure how we could test both with just one assert. An option would be to change the `testInput` to have only one `address`, and have it be the same for both test cases.

Reply:
I actually ask b/c I want us to explicitly test the fact we pass in multiple ids properly here...

two thoughts:

* each test case could add an `expectedAlowedIPIDs` field to be asserted on
* just break each test out into its own func (it _is_ just a table for two at this point ;) )

Reply:
great idea! should be good to go now.